lib/WUI/link_content/prusa_link_api.cpp

```diff
@@ -120,88 +174,41 @@ optional<ConnectionState> PrusaLinkApi::accept(const RequestParser &parser) cons
              */
             return std::visit([](auto upload) -> ConnectionState { return std::move(upload); }, std::move(upload));
         } else {
-            /*
-             * We do *not* use the uri with prefix removed. We need the safe
-             * transformation into a file name (removal of query params,
-             * forbidding of '..', etc).
-             */
-            /*
-             * TODO:
-             * We currently don't read the URI parameters. That is, we don't
-             * understand "?recursive=true" and therefore don't do recursive.
-             *
-             * The FileInfo handler doesn't know recursive yet either.
-             */
             static const auto prefix = "/api/files";
             static const size_t prefix_len = strlen(prefix);
-            char fname[FILE_PATH_BUFFER_LEN + prefix_len];
-            if (parser.uri_filename(fname, sizeof fname)) {
-                size_t len = strlen(fname);
-                if (prefix_len > len) {
-                    return StatusPage(Status::NotFound, parser.status_page_handling(), parser.accepts_json);
-                }
-
-                const char *fname_real = fname + prefix_len;
-
-                /*
-                 * The octoprint API gives special meaning to /local and
-                 * /sdcard. For us, everything lives in the USB (/usb). We
-                 * remap these. Nevertheless, we never _generate_ these /local
-                 * or such URIs, so we don't remap anything else but the "root".
-                 */
-                static const char *const roots[] = {
-                    "",
-                    "/",
-                    "/local",
-                    "/local/",
-                    "/sdcard",
-                    "/sdcard/",
-                };
-
-                for (size_t i = 0; i < sizeof roots / sizeof roots[0]; i++) {
-                    if (strcmp(fname_real, roots[i]) == 0) {
-                        fname_real = "/usb/";
-                        break;
-                    }
-                }
-
-                /*
-                 * Now, we make 100% sure the user won't get a file that's not
-                 * on the USB drive (eg. our xflash).
-                 */
-                if (strncmp(fname_real, "/usb/", 5) != 0) {
-                    return StatusPage(Status::Forbidden, parser.status_page_handling(), parser.accepts_json);
-                }
+            char filename[FILE_PATH_BUFFER_LEN + prefix_len];
```

Comment:
Am I right the parse_file_url outputs the filename here? Then it probably doesn't need the extra `prefix_len` any more?

I'm wondering about two things here, though.

* The buffer is not exactly small. And you now use two on the stack at the same time (the other one is inside the function). Would it make sense to declare that this parameter is not only output, but also the workspace?
* Alternatively, would it be cleaner to return some kind of `variant<array<char, ...>, error>`? I'm not sure if C++ guarantees some kind of return-placement optimization then or what, though, so maybe that could _also_ use two different buffers.